### PR TITLE
fix: skip reprocessing event if already seen; also bunch of small optimizations

### PR DESCRIPTION
**Pull request by ConjunctiveNormalForm**

### CODE REVIEW

File: src/aws_utils/cloudwatch_utils.rs [hunk 1]
```diff
@@ -1,120 +1,115 @@
...
 
-    impl Into<String> for DimensionName {
-        fn into(self) -> String {
-            match self {
-                DimensionName::Executor => EXECUTOR_DIMENSION.to_string(),
-            }
+impl From<DimensionName> for String {
+    fn from(dimension: DimensionName) -> Self {
+        match dimension {
+            DimensionName::Executor => EXECUTOR_DIMENSION.to_string(),
```
Comment (ConjunctiveNormalForm) [inline src/aws_utils/cloudwatch_utils.rs:26]: impl From gives you Into for free, the reverse not true

File: src/collectors/uniswapx_order_collector.rs [hunk 2]
```diff
@@ -44,21 +45,15 @@ impl FromStr for OrderType {
     }
 }
 
-impl ToString for OrderType {
-    fn to_string(&self) -> String {
+impl fmt::Display for OrderType {
```
Comment (ConjunctiveNormalForm) [inline src/collectors/uniswapx_order_collector.rs:48]: impl Display gives you ToString for free

File: src/strategies/keystore.rs [hunk 3]
```diff
@@ -95,6 +95,10 @@ impl KeyStore {
     pub fn len(&self) -> usize {
         self.keys.len()
     }
+
+    pub fn is_empty(&self) -> bool {
```
Comment (ConjunctiveNormalForm) [inline src/strategies/keystore.rs:99]: `pub fn len()` and `pub fn is_empty()` often go hand-in-hand to provide the same interface as std types

**Review by codyborn — APPROVED** [review]
More of a rust learning experience for me 😁 Looks good!

File: src/collectors/uniswapx_order_collector.rs (hunk 2 above)
Comment (zhongeric) [reply src/collectors/uniswapx_order_collector.rs:48]: nit: resolve Display import at top of file?

**Review by zhongeric — APPROVED** [review]
lgtm! do you have a different formatting setting?

**Pull request merged**